contracts/tokens/STFactory.sol

```diff
@@ -136,15 +137,18 @@ contract STFactory is ISTFactory, Ownable {
      * @param _logicContract Address of deployed module logic contract referenced from proxy
      * @param _upgradeData Data to be passed in call to upgradeToAndCall when a token upgrades its module
      */
-    function setLogicContract(string calldata _version, address _logicContract, bytes calldata _upgradeData) external onlyOwner {
+    function setLogicContract(string calldata _version, address _logicContract, bytes calldata _initializationData, bytes calldata _upgradeData) external onlyOwner {
         require(keccak256(abi.encodePacked(_version)) != keccak256(abi.encodePacked(logicContracts[latestUpgrade].version)), "Same version");
         require(_logicContract != logicContracts[latestUpgrade].logicContract, "Same version");
         require(_logicContract != address(0), "Invalid address");
+        require(_initializationData.length > 4, "Invalid Initialization");
+        require(_upgradeData.length > 4, "Invalid Upgrade");
```

Comment:
Should there be `_upgradeData` for all upgrades? I can see some minor upgrades not needing any upgrade data.

Reply:
In that case I was thinking we could still specify a no-op function (e.g. that just emits an Upgraded event or does nothing). This seemed "safer" than allowing an empty upgradeData which could be a mistake.